Userland/Libraries/LibWeb/HTML/HTMLHyperlinkElementUtils.cpp

```diff
@@ -555,16 +556,26 @@ DeprecatedString HTMLHyperlinkElementUtils::get_an_elements_target() const
     return "";
 }
 
-bool HTMLHyperlinkElementUtils::get_an_elements_noopener(StringView target) const
+DeprecatedString HTMLHyperlinkElementUtils::get_an_elements_rel() const
+{
+    if (auto rel = hyperlink_element_utils_rel(); !rel.is_empty())
+        return rel;
+
+    return "";
+}
+
+bool HTMLHyperlinkElementUtils::get_an_elements_noopener(StringView target, StringView rel) const
 {
     // To get an element's noopener, given an a, area, or form element element and a string target:
 
-    // FIXME: 1. If element's link types include the noopener or noreferrer
+    // 1. If element's link types include the noopener or noreferrer
```

Comment:
So the spec tells us we should split the `rel` attribute on ASCII whitespace:
https://html.spec.whatwg.org/multipage/links.html#linkTypes
> To determine which link types apply to a link, a, area, or form element, the element's rel attribute must be split on ASCII whitespace. The resulting tokens are the keywords for the link types that apply to that element.

So doing a substring search does not suffice because a site could say `rel="lolnoopenerlol"`, which is an invalid token, but would be accepted by this implementation.

I would remove the `get_an_elements_rel` function (get-an-element's-rel doesn't appear to be a concept in the spec, so we don't need a separate method with spec-like language for it). Then in `get_an_elements_noopener`, we can just do:

```c++
auto rel = hyperlink_element_utils_rel();
auto link_types = rel.view().split_view_if(Infra::is_ascii_whitespace);
```

(You'll need to `#include <LibWeb/Infra/CharacterTypes.h>` for the Infra method)

The rest of this method would then use e.g. `link_types.contains_slow("noopener"sv)` to check the types.